lib/Sema/ConstraintSystem.cpp

```diff
@@ -7308,27 +7308,8 @@ bool ConstraintSystem::isArgumentGenericFunction(Type argType, Expr *argExpr) {
 }
 
 bool ConstraintSystem::participatesInInference(ClosureExpr *closure) const {
-  if (getAppliedResultBuilderTransform(closure))
-    return true;
-
-  if (closure->hasSingleExpressionBody())
-    return true;
-
-  if (Options.contains(ConstraintSystemFlags::LeaveClosureBodyUnchecked))
-    return false;
-
-  if (closure->hasEmptyBody())
-    return false;
-
-  // If body is nested in a parent that has a function builder applied,
-  // let's prevent inference until result builders.
-  if (Options.contains(
-          ConstraintSystemFlags::
-              ClosuresInResultBuildersDontParticipateInInference)) {
-    return !isInResultBuilderContext(closure);
-  } else {
-    return true;
-  }
+  return getAppliedResultBuilderTransform(closure) ||
+         closure->hasSingleExpressionBody() || !closure->hasEmptyBody();
```

Comment:
Do we even need to check for `hasSingleExpressionBody` anymore? I think `!closure->hasEmptyBody()` should be enough

Reply:
Yes, looks like we can drop it.